### ISSUE TITLE
Reduce number of build operations emitted around artifact transforms

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#14674]
We currently emit build operations about transforms in the following places:

- `InitialTransformationNode.execute()` and `ChainedTransformationNode.execute()`
  -   type: `ArtifactTransformationStepBuildOperation`
  -   details: `ExecuteScheduledTransformationStepBuildOperationDetails`
  -   result: `ExecuteScheduledTransformationStepBuildOperationResult` (constant)
  -   category: `TRANSFORM`
  -   message: `"Transform $subjectName with $transformation"`
  -   progress name: `"Transforming $subjectName with $transformation"`
    - the `Transforming` prefix is used for filtering out artifact transform logs in Gradle Enterprise
- `TransformingAsyncArtifactListener.artifactAvailable()`
  -   type: `TransformationOperation`
  -   details: none
  -   result: none
  -   category: `UNCATEGORIZED`
  -   display name: `"Transform $initialSubject with $transformation"`
  -   progress name: yes, same as display name
- `DefaultTransformerInvocationFactory.createInvocation()`
  -   type: no specific type
  -   details: none
  -   result: none
  -   category: `UNCATEGORIZED`
  -   display name: `"$transformer $inputArtifact"`
  -   progress name: yes, same as display name

This is redundant, and I think we should reduce their types and numbers.

We can't delete the build operation in `TransformingAsyncArtifactListener`, as it is needed to put the thing into the `BuildOperationQueue`, but we can lose the progress name, so that it won't get reported to clients. We can also make it have a static display name to reduce unnecessary memory allocation.

The operation emitted from `DefaultTransformerInvocationFactory` is where we've gone through a bunch of cache layers, and are actually executing the transform (or loading it from cache at least). So this one should have the `TRANSFORM` category, the details and result objects attached to it, together with the nice name. We should probably also rename `ExecuteScheduledTransformationStepBuildOperation*` because it wouldn't be about scheduled transforms anymore, but any transform instead.

The one in `*TransformNode` can then be removed altogether.

---
cc: @gradle/execution

**Comment by wolfs** [comment]
The proposal sounds good. What I don't know if the console logging will trip up when we run a non-scheduled transform via `TransformingAsyncArtifactListener` (therefore nested below a task) and we emit a build operation with category `TRANSFORM`, which is supposed to be a top-level category. We should test this is we don't have a test already.

**Comment by yuriy-solodkyy** [comment]
Reducing the number of events Android Studio receives would still be helpful.

**Comment by wolfs** [comment]
I don't think any of the three build operations will go away, especially since we need the one in `*TransformationNode` in build scans. I think we also did some of the improvements described in this issue. Let's close it and re-open/create a new one if we want to change the build operations again.

**Comment by yuriy-solodkyy** [comment]
I don't think we needed reducing the number but rather being able to control what is returned at more granular level. I don't know what the current status is since Studio/IDEA avoid requesting even some basic events.
